### DISCUSSION TITLE
API to modify properties of existing connectivity objects

### DISCUSSION THREAD

**Discussion by ntolley** [jonescompneurolab/hnn-core#656] [Ideas]
After discussion with @rythorpe @jasmainak and @carolinafernandezp we've realized that it is currently very difficult to modify properties of the `net.connectivity` objects after they are made. I'm opening this discussion to work on the potential API for modifying these properties. 

I think the most parsimonious solution will be equipping the `_Connectivity` objects with `getter()` and `setter()` functions for the attributes we wish to modify (starting with basic ones right now like connection weight, lamtha, etc.).

Additionally it would be nice to equip them with method that filters the connections to subsets of gids, as well as changes the connection probability. I think these could actually non-interfering properties, where you can specify a set of `src_gids` and `target_gids`, and then the `probability` parameter changes the probability of a connection _within_ that subset.

**Comment by jasmainak** [comment]
Since `_Connectivity` already inherits from a `dict` isn't that already doable? 

I was thinking more of a `net.update_connection` method analogous to `net.add_connection`
